### ISSUE TITLE
Compiling fails on invalid nested archive

### ISSUE THREAD

**Issue by iskarian** [wabbajack-tools/wabbajack#972]
When compiling a list, Wabbajack seems to fail if 7z recognizes a file inside a mod archive as a nested archive but encounters errors in decompressing it. It fails despite the inner file not containing any needed files.

In particular, this is occurring with `Nemesis v0.83c-beta` for SSE, but it appears as if all recent versions include this file. It includes an intentionally problematic `tar` archive for unit testing: https://github.com/ShikyoKira/Project-New-Reign---Nemesis-Main/blob/master/test%20environment/Lib/test/testtar.tar

(The release is: https://github.com/ShikyoKira/Project-New-Reign---Nemesis-Main/releases/download/v0.83cSE-beta/Nemesis.Unlimited.Behavior.Engine.v0.83c-beta.rar)

It is unclear whether the issue is with the kind of error 7z is throwing (starting to decompress then failing, versus determining the file is not an archive at all), or if it is specifically related to nested extraction.

**Environment**
Win 10, Wabbajack 2.1.3.1, SSE

**Logs**
There are a bunch of these, which Wabbajack seems to handle fine:
```2917.02 - 7z.exe (21952) StdErr: ERROR: D:\Games\modlist\tools\zEdit\icudtl.dat
2917.02 - 7z.exe (21952) StdErr: Can not open the file as archive
```

But these it fails on:
```1550.43 - The contents of Nemesis.Unlimited.Behavior.Engine.v0.83c-beta.rar are being extracted to Nemesis.Unlimited.Behavior.Engine.v0.83c-beta.rar using 7zip.exe
1554.33 - Extracting Nemesis.Unlimited.Behavior.Engine.v0.83c-beta.rar - done
1556.37 - The contents of testtar.tar are being extracted to testtar.tar using 7zip.exe
1556.4 - 7z.exe (12252) StdErr: ERROR: D:\Games\Jabbawack\2.1.3.1\tmp_files\9bdd3ecf-a611-46d4-8a16-eea953da82fe\Nemesis_Engine\Lib\test\testtar.tar
1556.4 - 7z.exe (12252) StdErr: Can not open the file as archive
1556.4 - 7z.exe (12252) StdErr: The parameter is incorrect.
1556.41 - 7Zip returned an error while executing

1556.41 - Error while extracting D:\Games\Jabbawack\2.1.3.1\tmp_files\9bdd3ecf-a611-46d4-8a16-eea953da82fe\Nemesis_Engine\Lib\test\testtar.tar - Wabbajack.Common.StatusFeed.Errors._7zipReturnError: Exception of type 'Wabbajack.Common.StatusFeed.Errors._7zipReturnError' was thrown.
   at Wabbajack.Common.Utils.ErrorThrow(IException err) in c:\agent\_work\1\s\Wabbajack.Common\Utils.cs:line 162
   at Wabbajack.VirtualFileSystem.FileExtractor.ExtractAllWith7Zip(AbsolutePath source, IEnumerable`1 onlyFiles) in c:\agent\_work\1\s\Wabbajack.VirtualFileSystem\FileExtractor.cs:line 193
   at Wabbajack.VirtualFileSystem.FileExtractor.ExtractAll(WorkQueue queue, AbsolutePath source, IEnumerable`1 OnlyFiles) in c:\agent\_work\1\s\Wabbajack.VirtualFileSystem\FileExtractor.cs:line 39
1556.41 - Error while examining the contents of testtar.tar
1556.62 - Deleting D:\Games\Jabbawack\2.1.3.1\tmp_files\9bdd3ecf-a611-46d4-8a16-eea953da82fe
1557.57 - Error while examining the contents of Nemesis.Unlimited.Behavior.Engine.v0.83c-beta.rar
1557.58 - Compiler error - Wabbajack.Common.StatusFeed.Errors._7zipReturnError: Exception of type 'Wabbajack.Common.StatusFeed.Errors._7zipReturnError' was thrown.
   at Wabbajack.Common.Utils.ErrorThrow(Exception ex, String extraMessage) in c:\agent\_work\1\s\Wabbajack.Common\Utils.cs:line 150
   at Wabbajack.VirtualFileSystem.FileExtractor.ExtractAll(WorkQueue queue, AbsolutePath source, IEnumerable`1 OnlyFiles) in c:\agent\_work\1\s\Wabbajack.VirtualFileSystem\FileExtractor.cs:line 39
   at Wabbajack.VirtualFileSystem.VirtualFile.Analyze(Context context, VirtualFile parent, IExtractedFile extractedFile, IPath relPath, Int32 depth) in c:\agent\_work\1\s\Wabbajack.VirtualFileSystem\VirtualFile.cs:line 255
   at Wabbajack.Common.Utils.<>c__DisplayClass107_1`2.<<PMap>b__1>d.MoveNext() in c:\agent\_work\1\s\Wabbajack.Common\Utils.cs:line 585
--- End of stack trace from previous location where exception was thrown ---
   at Wabbajack.Common.Utils.PMap[TI,TR](IEnumerable`1 coll, WorkQueue queue, Func`2 f) in c:\agent\_work\1\s\Wabbajack.Common\Utils.cs:line 602
   at Wabbajack.VirtualFileSystem.VirtualFile.Analyze(Context context, VirtualFile parent, IExtractedFile extractedFile, IPath relPath, Int32 depth) in c:\agent\_work\1\s\Wabbajack.VirtualFileSystem\VirtualFile.cs:line 255
   at Wabbajack.VirtualFileSystem.VirtualFile.Analyze(Context context, VirtualFile parent, IExtractedFile extractedFile, IPath relPath, Int32 depth) in c:\agent\_work\1\s\Wabbajack.VirtualFileSystem\VirtualFile.cs:line 255
   at Wabbajack.VirtualFileSystem.Context.<>c__DisplayClass26_0.<<AddRoots>b__4>d.MoveNext() in c:\agent\_work\1\s\Wabbajack.VirtualFileSystem\Context.cs:line 107
--- End of stack trace from previous location where exception was thrown ---
   at Wabbajack.Common.Utils.<>c__DisplayClass107_1`2.<<PMap>b__1>d.MoveNext() in c:\agent\_work\1\s\Wabbajack.Common\Utils.cs:line 585
--- End of stack trace from previous location where exception was thrown ---
   at Wabbajack.Common.Utils.PMap[TI,TR](IEnumerable`1 coll, WorkQueue queue, Func`2 f) in c:\agent\_work\1\s\Wabbajack.Common\Utils.cs:line 602
   at Wabbajack.VirtualFileSystem.Context.AddRoots(List`1 roots) in c:\agent\_work\1\s\Wabbajack.VirtualFileSystem\Context.cs:line 96
   at Wabbajack.Lib.MO2Compiler._Begin(CancellationToken cancel) in c:\agent\_work\1\s\Wabbajack.Lib\MO2Compiler.cs:line 139
   at Wabbajack.Lib.ABatchProcessor.<Begin>b__46_0() in c:\agent\_work\1\s\Wabbajack.Lib\ABatchProcessor.cs:line 168
   at Wabbajack.MO2CompilerVM.Compile() in c:\agent\_work\1\s\Wabbajack\View Models\Compilers\MO2CompilerVM.cs:line 200
   at Wabbajack.CompilerVM.<>c__DisplayClass63_0.<<-ctor>b__19>d.MoveNext() in c:\agent\_work\1\s\Wabbajack\View Models\Compilers\CompilerVM.cs:line 182
```

**Comment by iskarian** [comment]
I just tested 7z's exit code semantics and it seems to be signaling `2` for both non-archives and invalid archives.

**Comment by halgari** [comment]
fixed in 2.1.3.2
